httpclient5/src/main/java/org/apache/hc/client5/http/nio/AsyncClientConnectionOperator.java

```diff
@@ -86,18 +113,35 @@
      * @param host the address of the opposite endpoint with TLS security.
      * @param attachment the attachment, which can be any object representing custom parameter
      *                    of the operation.
-     *
+     * @param callback the future result callback.
+     * @param context the execution context.
      * @since 5.2
      */
     default void upgrade(
             ManagedAsyncClientConnection conn,
             HttpHost host,
             Object attachment,
-            FutureCallback<ManagedAsyncClientConnection> callback) {
-        upgrade(conn, host, attachment);
+            FutureCallback<ManagedAsyncClientConnection> callback,
+            HttpContext context) {
```

Comment:
@ok2c I have changed the signature of this method since it has been introduce in 5.2 as well, hope it is OK, thanks!

Reply:
@reta Sure thing. The connection operators are considered internal APIs, so technically we can do with them whatever we please.